code/src/layouts/dashboard-tabs/DashboardTabs.tsx

```diff
@@ -0,0 +1,67 @@
+import { useLayoutEffect } from "react";
+import { useSearchParams } from "react-router-dom";
+
+import {
+  DASHBOARD_TAB_NAMES,
+  DashboardTabName,
+  dashboardTabs
+} from "@/layouts/dashboard-tabs/DashboardTabs.constants";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import cn from "@/utils/cn/cn";
+
+import "@/layouts/dashboard-tabs/DashboardTabs.scss";
+
+type TabSearchParam = DashboardTabName | null;
+
+const TAB_QUERY_KEY = "tab";
+
+const DashboardTabs = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const tabFromQuery = searchParams.get(TAB_QUERY_KEY) as TabSearchParam;
+
+  const handleSetActiveTab = (tab: DashboardTabName) => {
+    setSearchParams({ [TAB_QUERY_KEY]: tab });
+  };
+
+  useLayoutEffect(() => {
+    const tabNames = Object.values<TabSearchParam>(DASHBOARD_TAB_NAMES);
+
+    if (!tabNames.includes(tabFromQuery)) {
+      handleSetActiveTab(DASHBOARD_TAB_NAMES.USERS);
+    }
+  }, []);
+
+  const tabContent = dashboardTabs.find((tab) => tab.name === tabFromQuery);
+
+  return (
+    <AppBox className="dashboard-tabs">
+      <AppBox className="dashboard-tabs__label-container">
+        {dashboardTabs.map((tab) => (
+          <AppBox
+            key={tab.name}
+            onClick={() => handleSetActiveTab(tab.name)}
+            className={cn(
+              "dashboard-tabs__label-item",
+              tabFromQuery === tab.name && "dashboard-tabs__label-item--active"
+            )}
+          >
+            {tab.icon}
+            <AppTypography
+              component="span"
+              translationKey={tab.labelTranslationKey}
+            />
+          </AppBox>
+        ))}
```

Comment:
I suggest you move it above the return statement

Reply:
Moved. I also thought about creating separate `DashboardTabComponent`, I tried to do it, but it involves strange state splitting regarding `searchParams`, because we can use logic in either in parent or child component because search params can be accessed anywhere, it's like context in this case. What are you thoughts about it?